lib/active_record/connection_adapters/trilogy_adapter.rb

```diff
@@ -153,13 +153,99 @@ def discard!
         self.connection = nil
       end
 
+      # ActiveRecord 7.0 support
+      if ActiveRecord.version < ::Gem::Version.new('7.1.a')
+        def initialize(*args, **kwargs)
+          if kwargs.present?
+            args << kwargs.dup
+            kwargs.clear
+          end
+          # Turn  .new(config)  into  .new(nil, nil, nil, config)
+          3.times { args.unshift nil } if args.length < 4
+          super
+          if @connection
+            @verified = true
+            @raw_connection = @connection
+          end
+          # Ensure that we're treating prepared_statements in the same way that Rails 7.1 does
+          @prepared_statements = self.class.type_cast_config_to_boolean(
+            @config.fetch(:prepared_statements) { default_prepared_statements }
+          )
+        end
+
+        def connect!
+          verify!
+          self
+        end
+
+        alias_method :_original_active?, :active?
+        def active?
+          return false if connection&.closed?
+
+          _original_active?
+        end
```

Comment:
perhaps this change can be included in the `active?` method instead of just being overridden for AR 7.0 ?

Reply:
True.  Have placed this change in a separate commit in case there are any different ideas about how to implement the `.closed?` check.